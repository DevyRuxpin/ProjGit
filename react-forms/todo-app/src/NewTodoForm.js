import React, { useState } from 'react';

function NewTodoForm({ addTodo }) {
  const [task, setTask] = useState('');

  const handleSubmit = (evt) => {
    evt.preventDefault();
    addTodo({ task });
    setTask('');
  };

  const handleChange = (evt) => {
    setTask(evt.target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <div>
        <label htmlFor="task">Task:</label>
        <input
          id="task"
          name="task"
          value={task}
          onChange={handleChange}
          type="text"
          required
        />
      </div>
      <button type="submit">Add Todo</button>
    </form>
  );
}

export default NewTodoForm; 