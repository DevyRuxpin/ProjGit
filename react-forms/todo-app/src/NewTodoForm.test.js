import { render, screen, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

const mockAddTodo = jest.fn();

test('renders without crashing', () => {
  render(<NewTodoForm addTodo={mockAddTodo} />);
});

test('matches snapshot', () => {
  const { asFragment } = render(<NewTodoForm addTodo={mockAddTodo} />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders form input', () => {
  render(<NewTodoForm addTodo={mockAddTodo} />);
  
  expect(screen.getByLabelText(/task/i)).toBeInTheDocument();
  expect(screen.getByText(/add todo/i)).toBeInTheDocument();
});

test('can update form input', () => {
  render(<NewTodoForm addTodo={mockAddTodo} />);
  
  const taskInput = screen.getByLabelText(/task/i);

  fireEvent.change(taskInput, { target: { value: 'Buy groceries' } });

  expect(taskInput.value).toBe('Buy groceries');
});

test('calls addTodo with form data on submit', () => {
  render(<NewTodoForm addTodo={mockAddTodo} />);
  
  const taskInput = screen.getByLabelText(/task/i);
  const submitButton = screen.getByText(/add todo/i);

  fireEvent.change(taskInput, { target: { value: 'Buy groceries' } });
  fireEvent.click(submitButton);

  expect(mockAddTodo).toHaveBeenCalledWith({
    task: 'Buy groceries'
  });
});

test('clears form after submission', () => {
  render(<NewTodoForm addTodo={mockAddTodo} />);
  
  const taskInput = screen.getByLabelText(/task/i);
  const submitButton = screen.getByText(/add todo/i);

  fireEvent.change(taskInput, { target: { value: 'Buy groceries' } });
  fireEvent.click(submitButton);

  expect(taskInput.value).toBe('');
}); 