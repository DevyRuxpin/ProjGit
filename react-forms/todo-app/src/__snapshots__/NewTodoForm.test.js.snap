// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div>
      <label
        for="task"
      >
        Task:
      </label>
      <input
        id="task"
        name="task"
        required=""
        type="text"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
